### ISSUE TITLE
Update the author submission screen

### ISSUE THREAD

**Issue by ajrbyers** [openlibhums/janeway#4519]
Currently the author submission screen is spread across two columns which doesn't give us a lot of room to display information especially now that we are adding ROR and credit taxonomy information. This change would also be good for consistency as all other submission interfaces are single column.

- [ ] Sketch a new layout for the authors submission interface

**Comment by ajrbyers** [comment]
![Author Submission Single Col Sketch(1)](https://github.com/user-attachments/assets/06c83890-55c0-40bb-b4ff-fa0c39f56e5e)

**Comment by ajrbyers** [comment]
@MartinPaulEve @joemull @mauromsl @StephDriver @S-Haime @Astronaut-resting here is the sketch I put together for the making the author interface a single column. I feel like this is much easier to scan and understand than the current layout.

**Comment by StephDriver** [comment]
<img width="313" alt="image" src="https://github.com/user-attachments/assets/f647e2da-c262-4241-844c-02bb6df3e355">

I think something else is needed here to make this compatible with WCAG 2.4.4 link purpose in context.

see also [ROR Docs | Display](https://ror.readme.io/docs/display)

**Comment by ajrbyers** [comment]
> <img alt="image" width="313" src="https://private-user-images.githubusercontent.com/5330770/392024010-f647e2da-c262-4241-844c-02bb6df3e355.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMyMzk4MjcsIm5iZiI6MTczMzIzOTUyNywicGF0aCI6Ii81MzMwNzcwLzM5MjAyNDAxMC1mNjQ3ZTJkYS1jMjYyLTQyNDEtODQ0Yy0wMmJiNmRmM2UzNTUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDNUMTUyNTI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MmIzYjlhNGI1ZDEwNDBiNzRmZWEyNzBhMmE1OGJhNDZiNWZjNTE1MmMwYjc3YmQxMmYyZTllYjhlNWFkNzJlYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.WQf7bUkRst0rUT9_9g5nO4IWqE5FCXgzBwIP9CWv0j0">
> 
> I think something else is needed here to make this compatible with WCAG 2.4.4 link purpose in context.
> 
> see also [ROR Docs | Display](https://ror.readme.io/docs/display)

Ah nice to see they have this.


Ive uploaded a slightly tweaked version. It removes stuff about dragging and dropping which we don't really want.

**Comment by StephDriver** [comment]
Ok, I have a few ideas to add on here - I've not built up as much on the detail on inside the author cards, this is more about the general flow of the cards on the page, thinking of the user interactions, and asking them for one thing at a time, while making it really clear which is the correspondence author.

Colours are only illustrative, to show which are same and which different, this is about the user journey through the page, not the colour details.

## TLDR
- I think his page can be simplified into a list of author cards, with an add card at the bottom.
- All methods for adding authors are in the same add card.  When there are no authors already, this is all that is shown.
- The main/correspondence author is shown on the author's card, not chosen after, this highlights clearly in the list which author is there for correspondence.

## Designs

### No Authors Yet

<img width="851" alt="image" src="https://github.com/user-attachments/assets/2a1edae0-aa15-4c1e-a25f-c8bf12b05c23">

### One Author only
If there is only one author, then this should default to the correspondence author.

<img width="859" alt="image" src="https://github.com/user-attachments/assets/7b97387a-2e94-485e-863d-1982eb860c09">

### Once there are several authors
The correspondence author could be changed, only once there are multiple authors from which to choose.

<img width="473" alt="image" src="https://github.com/user-attachments/assets/562af803-7127-4c89-b8f9-8101a97d7377">

**Comment by ajrbyers** [comment]
@StephDriver this is great! As discussed on Discord lets use our time on Thursday to iterate this a bit.

**Comment by StephDriver** [comment]
> I really like the direction this is heading in. It's a good candidate for the design meeting tomorrow.
> 
> The only suggestion I have is to use single-click buttons to change the correspondence author, like the move up and move down buttons. All non-primary authors could have a button called "Make correspondence author", and the correspondence author could have a label instead of a button that says "Correspondence author", perhaps with an appropriate icon like a star.

Happy with this... my thoughts on the main author were:
1. if main = correspondence, then call it correspondence
2. make it obvious at a glance which is the correspondence author, at the point where their details are, not as a separate thing
3. make it self-evident you can only have one correspondence author, and so to have a different one you have to change the one you already have.
4. if there is only one author, then that is the correspondence author by default.  You only have to think about this if and when you have more than one author.
5. I did wonder about whether the correspondence author should default to 'self' if you select self, simply because as the person filling this in it was likely you were the correspondence author, but I think how to make that work without complicating things other than as an option within the 'add self as author' modal, and sketching modals felt beyond time and scope.

**Comment by joemull** [comment]
> make it self-evident you can only have one correspondence author, and so to have a different one you have to change the one you already have.

OK, I see this, but then what does the interface look like when you select "Change" on the correspondence author? Is it a dropdown? What do the options look like? Are they the author's names?

![image](https://github.com/user-attachments/assets/b953efe9-e859-4586-9057-237cfb165982)

Also with regards to copy, I think we should maintain the label "Correspondence Author" and make sure this is consistent with labels elsewhere in the interface. I think most places use "Correspondence Author" but some I know of use "Primary", and we should get rid of the discrepancy if we are thinking this carefully about what to call it in this interface.

**Comment by Astronaut-resting** [comment]
Few questions/comments

Totally agree that this page had a lot of redundancies and this is a great direction! 

1) On that note, do we need to provide 3 methods of adding an author? i am not sure of the function of having "add self as author". Can't this just come under "add author" generally? 
Having it set out like this doesn't make it clear that what we are providing is three different methods of doing the same thing (we could be asking people to complete the search first, followed by the adding). 
Therefore, maybe we could just distinguish between "search author from Janeway database" and "add author manually". 
(in training, AB has to explain that exact match pertains to the Janeway database rather than an ORCid match, so we should try to make that clear through wording) 

2) As a user, I wouldn't be sure of the purpose of "moving up" and "moving down". Is this _just_ changing the order on this screen? or does it have something to do with whom is primary? I think not, because the correspondence label is a separate thing.

**Comment by ajrbyers** [comment]
> Few questions/comments
> 
> Totally agree that this page had a lot of redundancies and this is a great direction!
> 
>     1. On that note, do we need to provide 3 methods of adding an author? i am not sure of the function of having "add self as author". Can't this just come under "add author" generally?
>        Having it set out like this doesn't make it clear that what we are providing is three different methods of doing the same thing (we could be asking people to complete the search first, followed by the adding).
>        Therefore, maybe we could just distinguish between "search author from Janeway database" and "add author manually".
>        (in training, AB has to explain that exact match pertains to the Janeway database rather than an ORCid match, so we should try to make that clear through wording)
> 
>     2. As a user, I wouldn't be sure of the purpose of "moving up" and "moving down". Is this _just_ changing the order on this screen? or does it have something to do with whom is primary? I think not, because the correspondence label is a separate thing.

I feel like the text in my version covers this though the copy might want to be re-written.

**Comment by Astronaut-resting** [comment]
![IMG_5430](https://github.com/user-attachments/assets/a33032c4-bfd2-4450-a53f-8ca949a3c16f)

**Comment by Astronaut-resting** [comment]
Ah, I did not realise that this was a default situation and that's why you have it rather than just a more universal "add author". I could not pick that up from Steph's diagram. Only when I went into Janeway itself did I understand because the copy reads:
"By default, your account is the owner of this submission, but is not an Author on record. You can add yourself using the button below."

Hmm. This is a conundrum! My concern is that we are asking them too many things at once in this one block. Are we asking them to choose between two columns where it is
Search for author OR add author manually 
AND
Add yourself as an author OR add another person as an author 

Maybe that level of complexity is fine. It is still an improvement

**Comment by StephDriver** [comment]
> Hmm. This is a conundrum! My concern is that we are asking them too many things at once in this one block. Are we asking them to choose between two columns where it is Search for author OR add author manually AND Add yourself as an author OR add another person as an author

I am not a fan of the 'add self as author' being on the same level as the 'add author' but I couldn't think of something better so kept it.

I do think if you are currently listed as an author this should then be hidden as an option. In general, all options should be hidden from contexts where they don't do anything (add self if self already present, move up if at top, move down if at bottom, etc).

**Comment by ajrbyers** [comment]
It is hidden if you’re an author.

**Comment by StephDriver** [comment]
out of interest, can you submit and be the correspondence person without being an author at all?

**Comment by Astronaut-resting** [comment]
> Do you think its not easier to give the author the option to easily add themselves rather than ask them to search for themselves? Most journals have the setting to make the submitted an author automatically so most don't see this.

Thinking about it probably the best solution to my mind is to have the account automatically there as an author on the screen, with the ability to remove it. 
THEN ask them to choose between searching additional authors or manually adding them 
This has the added benefit of providing an example on the screen of what an added author looks like (from which they can add further or subtract. 

![IMG_5431 2](https://github.com/user-attachments/assets/1fb7ff3f-f848-4ac8-b92c-3dd612d42413)

**Comment by StephDriver** [comment]
> 2\. As a user, I wouldn't be sure of the purpose of "moving up" and "moving down". Is this _just_ changing the order on this screen? or does it have something to do with whom is primary? I think not, because the correspondence label is a separate thing.

I was musing about making the order really clear, like numbering them, not just having top to bottom.  So its clear 'first author' etc.

<img width="717" alt="image" src="https://github.com/user-attachments/assets/32c8338a-d27c-4a65-981d-fe9520daa5a6">

**Comment by joemull** [comment]
We had a thorough discussion of this in the design meeting, and came to consensus about a number of things:

* Unify the features for adding self, searching for existing author, and adding a new author in a single card that stays at the top of the page as authors are added, like this:

  ![image](https://github.com/user-attachments/assets/fad45ce6-6ca4-4aa7-8cd3-c3ea8175a1bc)

  Caveat: after the meeting, I read @Astronaut-resting 's comments for the first time about automatically adding self as author. I like this idea. Did we discuss it in the meeting and I missed it? If not, can we consider it?

* The author search will also check the ORCID API and populate an author if available. The help text will make it clear you can search with ORCIDs.

* The author cards appear after the first card, like in Steph's design, but with a simple number for the label, not "First Author". We can use `ol` for the markup so that it's clear to screen readers that there is ordering happening.

* No Edit button for the card, just for each ROR

* Delete asks you to confirm deletion, either on a separate page or with a modal

* The button for changing author order opens a space below it for selecting a different author

* The Add CRediT button also opens a space below for selecting a role

* The current CRediT roles need buttons for removing them

* Four small buttons (no text) for the author order controls, with the conventional icons, which allow the user to move to top, move up, move down, or move to bottom

Here's an approximation (note a few differences with what's described above!):

![image](https://github.com/user-attachments/assets/ab9c3c55-fffa-4a54-9df3-1a25e77a86c6)

**Comment by ajrbyers** [comment]
> Caveat: after the meeting, I read @Astronaut-resting 's comments for the first time about automatically adding self as author. I like this idea. Did we discuss it in the meeting and I missed it? If not, can we consider it?

There is a setting for this already - I think we should toggle it to on by default though.

**Comment by StephDriver** [comment]
I'm closing #4306 as it is now covered by this work.
- https://github.com/openlibhums/janeway/issues/4306
That issue was about it being confusing on this page for a 'search' button should not have 'add' functionality.  All the design discussions above have got rid of that confusion.
